flixel/util/FlxDirections.hx

```diff
@@ -0,0 +1,117 @@
+package flixel.util;
+
+/**
+ * Uses bit flags to create a list of orthogonal directions. useful for
+ * many `FlxObject` features like `allowCollisions` and `touching`.
+ * @since 4.10.0
+ */
+@:enum abstract FlxDirections(Int) from Int to Int
+{
+	/** Generic value for "left". */
+	var LEFT    = 0x0001;
+	/** Generic value for "right". */
+	var RIGHT   = 0x0010;
+	/** Generic value for "up". */
+	var UP      = 0x0100;
+	/** Generic value for "down". */
+	var DOWN    = 0x1000;
+	/** Special-case constant meaning no directions. */
+	var NONE    = 0x0000;
+	
+	/**
+	 * Directions values can't be constructed from other direction values, due to a bug in haxe.
+	 * https://github.com/HaxeFoundation/haxe/issues/10237
+	 * We'll just define all these using literal hex values, otherwise they can't be used as
+	 * default values for function arguments.
+	 */
+	
+	/** Special-case constant meaning "up". */
+	var CEILING = 0x0100;// UP;
+	/** Special-case constant meaning "down" */
+	var FLOOR   = 0x1000;// DOWN;
+	/** Special-case constant meaning "left" and "right". */
+	var WALL    = 0x0011;// LEFT | RIGHT;
+	/** Special-case constant meaning any, or all directions. */
+	var ANY     = 0x1111;// LEFT | RIGHT | UP | DOWN;
+	
+	/**
+	 * Returns true if this contains all of the supplied flags.
+	 */
+	public inline function hasAll(dir:FlxDirections):Bool
+	{
+		return this & dir == dir;
+	}
+	
+	/**
+	 * Returns true if this contains any one of the supplied flags.
+	 */
+	public inline function hasAny(dir:FlxDirections):Bool
```

Comment:
I'm not sure this is very useful, I think I preferred the simplicity of having just one `has()` (which is now called `hasAll()` I guess). It seems like you're already using it wrong too (no need for this in `toString()`, since you're always just checking against a single direction)?